frontend/src/store.ts

```diff
@@ -0,0 +1,48 @@
+import { configureStore, createSlice, PayloadAction } from '@reduxjs/toolkit';
+
+interface State {
+  isApiKeyModalOpen: boolean;
+  apiKey: string;
+  isMenuOpen: boolean;
+}
+
+const initialState: State = {
+  isApiKeyModalOpen: false,
+  apiKey: '',

```

Comment:
These two data `isApiKeyModalOpen` and `apiKey`  see tightly correlated. 
If API key is not present => show modal.
If API key present => dont show the modal.

 If this is the case then `isApiKeyModalOpen` is derived data and we shouldnt be storing that in redux. We should be writing a selector to derive it from `apiKey`
 
` isApiKeyModalOpen = ()=> !!apiKey`

Reply:
Dont understand this. Currently on page load apiKey is going to just be a blank string and we can choose to make isApiKeyModalOpen true after testing so the user is greeted with it when opening the page.

In the future when authentication is implemented this will probably change